perf/reader_test.go

```diff
@@ -298,6 +302,82 @@ func TestReadRecord(t *testing.T) {
 	}
 }
 
+func TestMute(t *testing.T) {
+	prog, events := mustOutputSamplesProg(t, 5)
+	defer prog.Close()
+	defer events.Close()
+
+	rd, err := NewReader(events, 4096)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rd.Close()
+
+	// Reader is already unmuted by default. It should be idempotent.
+	if err = rd.Unmute(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Write a sample. The reader should read it.
+	ret, _, err := prog.Test(make([]byte, 14))
+	if err != nil || ret != 0 {
+		t.Fatal("Can't write sample")
+	}
+	if _, err := rd.Read(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Mute. No notification should trigger.
+	if err = rd.Mute(); err != nil {
+		t.Fatal(err)
+	}
+	errChan := make(chan error)
+	go func() {
+		// Read one notification then send any errors and exit.
+		_, err := rd.Read()
+		errChan <- err
+		close(errChan)
```

Comment:
I think it should be OK to skip the close, then we have to handle one less case in the select statement below.